board/batocera/fsoverlay/etc/init.d/S02resize

```diff
@@ -40,44 +41,95 @@ function textoutput()
 # only when resizing is wanted
 if grep -qE '^[ ]*autoresize[ ]*=[ ]*true[ ]*$' "${BOOTCONF}"
 then
-
     # Preparing text arrays
     arr=(
-	"Removing trigger............" "Pending"
-	"Syncing disk data..........." "Pending"
-	"Checking disk table........." "Pending"
-	"Resizing partition.........." "Pending"
-	"Aligning /userdata.........." "Pending"
-	"Checking /userdata.........." "Pending"
-	"Resizing /userdata.........." "Pending"
+    "Syncing disk data..........." "Pending"
+    "Checking disk table........." "Pending"
+    "Aligning partition.........." "Pending"
+    "Resizing partition.........." "Pending"
+    "Checking /userdata.........." "Pending"
+    "Resizing /userdata.........." "Pending"
+    "Informing the Kernel........" "Pending"
+    "Syncing disk data..........." "Pending"
     )
 
     # --- BEGIN RESIZE ---
     # /userdata partition
     PART=$(batocera-part "share_internal")
+    echo "Partition = $PART" >> $log
     # boot disk
     DISK=$(batocera-part prefix "${PART}")
+    echo "Disk = $DISK" >> $log
 
     # only for ext4
     PARTTYPE=$(blkid "${PART}" | sed -e s+'^.* TYPE="\([^"]*\)\".*'+'\1'+)
     test "${PARTTYPE}" != "ext4" && exit 0
 
-    # fix any minor issues, such as gpt header not at end of disk
+    # get the size of the disk in bytes
+    # this may change if fdisk is updated
+    disk_size_sector=$(fdisk -l $DISK | grep 'Disk /dev' | awk '{print $3}')
+    echo "Disk Sector Size = $disk_size_sector" >> $log
+    
+    # get the sector size of the disk in bytes
+    sector_size=$(fdisk -l $DISK | grep 'Logical sector size' | awk '{print $4}')
+    
+    # convert the disk size to bytes
+    echo "Sector Size = $sector_size" >> $log
+    disk_size=$((disk_size_sector * sector_size))
+    echo "Disk Size in bytes = $disk_size" >> $log
+    
+    # calculate the size of the secondary GPT table
+    # usually 16512 bytes or 33 x 512 bytes
+    # we will double the size to be sure
+    secondary_gpt_size=$((2 * 33 * sector_size))
+    echo "GPT size = $secondary_gpt_size" >> $log
+    
+    # calculate the new size of the partition, taking into account the secondary GPT table
+    new_partition_size=$((disk_size - secondary_gpt_size))
+    echo "New partition end in bytes = $new_partition_size" >> $log
+    
+    # convert the new end sector to megabytes, rounding down to the nearest integer
+    new_end_size=$((new_partition_size / 1000000))
+    echo "New partition end in MB = $new_end_size" >> $log
+
     # remove the trigger
     mount -o remount,rw /boot && sed -i -e s+'^[ ]*autoresize'+'#autoresize'+ "${BOOTCONF}" && mount -o remount,ro /boot
 
     # textoutput "Message" "percentage" "command call" 
```

Comment:
```
# Array is "TEXT" "STATUS" and starts with zero, Usage:
# textoutput "Message [odd number]" "percentage [0-100]" "executed command call"
for i in 1 3 5 7 9 11 13 15; do
   case $i in
        1) 
           # initial disk sync
           echo "Step $i: Sync" >> "$log"
           textoutput $i 10 "sync";;
        3) 
           # move backup GPT data structures to the end of the disk
           echo "Step $i: Moving 2nd GPT table to the end of the disk" >> "$log"
           textoutput $i 20 "sgdisk -e ${DISK}";;
        5) 
           # checking the alignment before resizing the partition
.....
```

Reply:
sorry i did not see your messages @crcerror yes you're right - i will adjust when i can.